src/components/button-bar/button-bar.test.tsx

```diff
@@ -0,0 +1,192 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+
+import Button from "../button";
+import Icon from "../icon";
+import IconButton from "../icon-button";
+import ButtonBar from "./button-bar.component";
+
+describe("When ButtonBar children are Button components", () => {
+  it("should render the Buttons as expected", () => {
+    render(
+      <ButtonBar>
+        <Button aria-label="button-in-bar">foo</Button>
+        <Button aria-label="button-in-bar">bar</Button>
+        <Button aria-label="button-in-bar">baz</Button>
+      </ButtonBar>
+    );
+
+    const buttonChildren = screen.getAllByRole("button", {
+      name: "button-in-bar",
+    });
+
+    expect(buttonChildren).toHaveLength(3);
+  });
+
+  it("should render the Buttons with icons as expected when the 'iconType' prop is set", () => {
+    render(
+      <ButtonBar>
+        <Button aria-label="button-in-bar" iconType="bin">
+          foo
+        </Button>
+        <Button aria-label="button-in-bar" iconType="bin">
+          bar
+        </Button>
+        <Button aria-label="button-in-bar" iconType="bin">
+          baz
+        </Button>
+      </ButtonBar>
+    );
+
+    const buttonChildren = screen.getAllByRole("button", {
+      name: "button-in-bar",
+    });
+    const icons = screen.getAllByTestId("icon");
+
+    expect(buttonChildren).toHaveLength(3);
+    expect(icons).toHaveLength(3);
+  });
+
+  it("should render the Buttons with 'size' prop set to 'medium'", () => {
+    render(
+      <ButtonBar>
+        <Button aria-label="button-in-bar">foo</Button>
+        <Button aria-label="button-in-bar">bar</Button>
+        <Button aria-label="button-in-bar">baz</Button>
+      </ButtonBar>
+    );
+
+    const buttonChildren = screen.getAllByRole("button", {
+      name: "button-in-bar",
+    });
+
+    buttonChildren.forEach((button) => {
+      expect(button).toHaveStyle("font-size: var(--fontSizes100)");
+      expect(button).toHaveStyle("min-height: 40px");
```

Comment:
**suggestion(non-blocking):** `toHaveStyle` accepts an object so you could put these into one assertion. I'm happy with either way. It's just whichever way we decide we want to do it going forward.

```suggestion
      expect(button).toHaveStyle({
        fontSize: "var(--fontSizes100)",
        minHeight: "40px",
      });
```